web-frontend/js/api-calls.js

```diff
@@ -1,30 +1,75 @@
-var curr_lang = "python3";
-let url = "http://localhost:10100/api/v1/";
+var curr_lang;
+let url = "https://runner.fly.dev:10100/api/v1/";
```

Comment:
nit/non-blocking especially if this was intentional: the path https://runner.fly.dev/api/v1/ should work too. No port specification is needed. But probably easier to have the port if you want to swap between local and remote servers.